components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue

```diff
@@ -129,21 +129,25 @@ const disabled = computed(() => {
   return Number(balance.value) <= Number(props.nft.price)
 })
 
+const openCompletePurcahseModal = () => {
+  shoppingCartStore.setItemToBuy(nftToShoppingCardItem(props.nft))
+  preferencesStore.setCompletePurchaseModal({
+    isOpen: true,
+    mode: 'buy-now',
+  })
+}
+
 function onClick() {
   if (!connected.value) {
-    openConnectWalletModal(instance)
-    return
-  }
-
-  if (shoppingCartStore.isItemInCart(props.nft.id)) {
-    openShoppingCart(instance)
-  } else {
-    shoppingCartStore.setItemToBuy(nftToShoppingCardItem(props.nft))
-    preferencesStore.setCompletePurchaseModal({
-      isOpen: true,
-      mode: 'buy-now',
+    openConnectWalletModal(instance, {
+      onConnect: openCompletePurcahseModal,
+      closeAfterConnect: true,
     })
+    return
   }
+  shoppingCartStore.isItemInCart(props.nft.id)
+    ? openShoppingCart(instance)
+    : openCompletePurcahseModal()
 }
```

Comment:
The onConnect callback should be this one, right? 
```javascript
() => { 
shoppingCartStore.isItemInCart(props.nft.id)
    ? openShoppingCart(instance)
    : openCompletePurcahseModal() 
}
```

IMO we could make a composable to reuse the logic of connecting wallet.

Reply:
no. The logic is as follows:

if the item is in the cart => open the cart regardless of the wallet status
otherwise:
connected => open confirmation modal
not connected > open wallet => success connect => confirmation modal

Reply:
Alright, this old logic changed after the new commits.